crates/bevy_ecs/src/spawn.rs

```diff
@@ -311,15 +322,23 @@ impl<R: Relationship, B: Bundle> BundleEffect for SpawnOneRelated<R, B> {
     }
 }
 
-impl<R: Relationship, B: Bundle> DynamicBundle for SpawnOneRelated<R, B> {
+unsafe impl<R: Relationship, B: Bundle> DynamicBundle for SpawnOneRelated<R, B> {
     type Effect = Self;
 
-    fn get_components(
-        self,
+    unsafe fn get_components(
+        _ptr: *const Self,
         func: &mut impl FnMut(crate::component::StorageType, bevy_ptr::OwningPtr<'_>),
-    ) -> Self::Effect {
-        <R::RelationshipTarget as RelationshipTarget>::with_capacity(1).get_components(func);
-        self
+    ) {
+        let target = <R::RelationshipTarget as RelationshipTarget>::with_capacity(1);
+        let target = ManuallyDrop::new(target);
+        let target_ptr = &raw const target;
+        <R::RelationshipTarget as DynamicBundle>::get_components(target_ptr.cast::<R::RelationshipTarget>(), func);
+    }
+
+    unsafe fn apply_effect(ptr: *const Self, entity: &mut EntityWorldMut) {
+        // SAFETY: The caller must ensure that the `ptr` must be valid and aligned.
```

Comment:
This safety comment is wrong, the documentation for `apply_effect` states that `ptr` is not necessarily aligned